prow/bugzilla/client.go

```diff
@@ -174,20 +177,58 @@ func (c *client) AddPullRequestAsExternalBug(id int, org, repo string, num int)
 			BugIDs: []int{id},
 			ExternalBugs: []NewExternalBugIdentifier{{
 				Type: "https://github.com/",
-				ID:   fmt.Sprintf("%s/%s/pull/%d", org, repo, num),
+				ID:   pullIdentifier,
 			}},
 		}},
 	}
 	body, err := json.Marshal(rpcPayload)
 	if err != nil {
-		return fmt.Errorf("failed to marshal JSONRPC payload: %v", err)
+		return false, fmt.Errorf("failed to marshal JSONRPC payload: %v", err)
 	}
 	req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/jsonrpc.cgi", c.endpoint), bytes.NewBuffer(body))
 	if err != nil {
-		return err
+		return false, err
 	}
 	req.Header.Set("Content-Type", "application/json")
 
-	_, err = c.request(req, logger)
-	return err
+	resp, err := c.request(req, logger)
+	if err != nil {
+		return false, err
+	}
+	var response struct {
+		Error *struct {
+			Code    int    `json:"code"`
+			Message string `json:"message"`
+		} `json:"error,omitempty"`
+		ID     string `json:"id"`
+		Result *struct {
+			Bugs []struct {
+				ID      int `json:"id"`
+				Changes struct {
+					ExternalBugs struct {
+						Added   string `json:"added"`
+						Removed string `json:"removed"`
+					} `json:"ext_bz_bug_map.ext_bz_bug_id"`
+				} `json:"changes"`
+			} `json:"bugs"`
+		} `json:"result,omitempty"`
+	}
+	if err := json.Unmarshal(resp, &response); err != nil {
+		return false, fmt.Errorf("failed to unmarshal JSONRPC response: %v", err)
+	}
+	if response.Error != nil {
+		return false, fmt.Errorf("JSONRPC error %d: %v", response.Error.Code, response.Error.Message)
+	}
+	if response.ID != rpcPayload.ID {
+		return false, fmt.Errorf("JSONRPC returned mismatched identifier, expected %s but got %s", rpcPayload.ID, response.ID)
+	}
+	changed := false
+	if response.Result != nil {
+		for _, bug := range response.Result.Bugs {
+			if bug.ID == id {
+				changed = strings.Contains(bug.Changes.ExternalBugs.Added, pullIdentifier)
```

Comment:
is it possible for bugzilla to return the same id twice?  if so, make this `changed ||= ...` because I don't trust bugzilla very far...

Reply:
Technically yes since you can request more than one change at once but we never do, can update regardless

Reply:
Why not return `true` immediately?